packages/bcer-api/app/src/location/location.service.ts

```diff
@@ -432,11 +432,41 @@ export class LocationService {
 
     /**
    * Delete locations with ids,
-   * This is a soft-delete by manipulating the location status.
+   * This is a hard delete by location ID, fully removing the location and associated resources
    * @param locationIds 
    */
-     async hardDeleteLocation(locationId: string){
-      await this.locationRepository.delete(locationId);
+     async hardDeleteLocation(locationId: string): Promise<void>{
+      const connection = getConnection();
+      const queryRunner = connection.createQueryRunner();
+
+      const [location] = await queryRunner.query(`SELECT * FROM LOCATION WHERE "id"='${locationId}'`);
+      const noi = await queryRunner.query(`SELECT * FROM NOI WHERE "id"='${location.noiId}'`);
+      const locationManufacturesJoin = await queryRunner.query(`SELECT * FROM location_manufactures_manufacturing WHERE "locationId"='${locationId}'`);
+      const locationProductsJoin = await queryRunner.query(`SELECT * FROM LOCATION_PRODUCTS_PRODUCT WHERE "locationId"='${locationId}'`);
+      const productSold = await queryRunner.query(`SELECT * FROM PRODUCT_SOLD WHERE "locationId"='${locationId}'`);
+      const salesReport = await queryRunner.query(`SELECT * FROM SALESREPORT WHERE "locationId"='${locationId}'`);
+
+      await queryRunner.startTransaction();
+      try {
+
+        await Promise.all(locationManufacturesJoin.map(entity => queryRunner.manager.delete('location_manufactures_manufacturing', { manufacturingId: entity.productId })));
```

Comment:
I believe `location_manufactures_manufacturing` has a `locationId`, so should be able to just do 

```
await queryRunner.manager.delete('location_manufactures_manufacturing', { locationId })
```

instead of deleting each individually?

Reply:
Same for `location_products_product`/`salesreport`/`product_sold`